A74-xds-config-tears.md

```diff
@@ -0,0 +1,309 @@
+  // (a) there was an error and we did not already have a valid
+  // resource or (b) the resource does not exist.
+  // For aggregate clusters, the aggregate cluster will be first in
+  // the list, followed by the list of leaf clusters.  For other types
+  // of clusters, the list will have only one element.
+  struct ClusterConfig {
+    // Cluster name and resource.
+    std::string cluster_name;
+    XdsClusterResource cluster;
+    // Endpoint info.  If there was an error, endpoints will be null
+    // and resolution_note will be set.
+    XdsEndpointResource endpoints;
+    std::string resolution_note;
+  };
+  std::map<std::string, absl::StatusOr<std::vector<ClusterConfig>>> clusters;
+};
+```
+
+Note that the watcher's OnError() and OnResourceDoesNotExist()
+methods will be invoked only for problems with the listener and route
+configuration resources.  For cluster-specific errors, we need to
+retain the existing behavior, which is that it causes a problem for the
+individual cluster but does not affect traffic sent to any other cluster.
+Therefore, when the XdsDependencyManager sees an error on a cluster for
+which it did not previously have data, or if it sees a does-not-exist
+notification on a cluster, it will report that by returning a non-OK
+status for the cluster in the XdsConfig.  Similarly, if there is a problem
+obtaining endpoint data from EDS or DNS, the XdsDependencyManager will
+report that problem via the cluster's resolution_note field.
+
+The xds resolver will pass the returned XdsConfig to the LB policies via
+an attribute, so that the LB policies can use the cluster data.  Note
+that we cannot simply pass the data via the cds LB policy's JSON config,
+because we need this to work in cases where the cluster is determined
+dynamically, such as where a route uses a ClusterSpecifierPlugin.
+
+In order to handle such cases, the XdsDependencyManager will also provide
+an API to subscribe to a particular cluster resource whose name was
+not present in the route configuration.  The xds resolver will pass the
+XdsDependencyManager to the LB policies via an attribute, so that the
+cds LB policy can use this API.  The API will return some subscription
+handle that the caller can release when they are no longer interested
+in the cluster subscription.
+
+Note that in the XdsConfig, the cluster map will contain an entry for
+every cluster that is either referenced in the route configuration or
+for which a dynamic subscription was registered.  If a cluster is
+present only underneath an aggregate cluster, it will be present in the
+list for that aggregate cluster, but it will not have its own top-level
+entry.
+
+#### Changes in cds LB policy
+
+Instead of starting a watch on the XdsClient to get the cluster data,
+the cds LB policy will get its configuration by lookup up the cluster
+name in the XdsConfig passed to it via an attribute.
+
+To address the case where the cluster is determined dynamically (e.g., a
```

Comment:
Is this part about existing functionality? I don't see any existing mention of `ClusterSpecifierPlugin` in this repo.

Reply:
I think it would be good to have a note in this part about dynamic clusters that this is related to a non-public feature, don't think they missed something in some other gRFC.

Reply:
Done.